orbit/utils/simulation.py

```diff
@@ -86,14 +87,73 @@ def make_seasonality(series_len, seasonality, order=3, duration=1, scale=.05, ty
     return seas
 
 
+def make_regression(series_len, coefs, loc=0.0, scale=0.5, cov=None,
+                    noise_scale=1.0, relevance=1.0, sparsity=0.2, seed=1):
+    """ Module to generate multiplicative time-series with trend, seasonality and regression components
+    Parameters
+    ----------
+    series_len: int
+    coefs: 1-D array_like
+        values used as regression coefficients
+    loc: float
+        location parameter to generate regressor values
+    scale: float
+        scale parameter to generate regressor values
+    cov: 2-D array_like, of shape (num_of_regressors, num_of_regressors)
+        covariance of regressors in log unit scale
+    relevance: float
+        0 to 1; smaller value indicates less number of total useful regressors
+    sparsity: float
+        0 to 1 to control probability (= 1 - sparsity) at time t of a regressor value > 0
+    seed: int
+    Notes
+    ------
+        Some ideas are from https://scikit-learn.org/stable/auto_examples/linear_model/plot_bayesian_ridge.html
+        and https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.BayesianRidge.html#sklearn.linear_model.BayesianRidge
+    """
+
+    num_of_regressors = len(coefs)
+    if (relevance > 0.0) and (relevance < 1.0):
+        num_of_irr_coefs = int(num_of_regressors * (1 - relevance))
+        coefs = np.copy(coefs)
+        irr_idx = np.random.choice(num_of_regressors, num_of_irr_coefs, replace=False)
+        coefs[irr_idx] = 0.0
+
+    if cov is None:
+        x = np.random.default_rng(seed).normal(
+            loc,
+            scale,
+            series_len * num_of_regressors
+        ).reshape(series_len, -1)
+    else:
+        x = np.random.default_rng(seed).multivariate_normal(
+            np.array([loc] * num_of_regressors, dtype=np.float64),
+            cov,
+            series_len
+        )
+    # control probability of regression kick-in
+    if (sparsity > 0.0) and (sparsity < 1.0):
+        z = np.random.default_rng(seed).binomial(
+            1,
+            1 - sparsity,
+            series_len * num_of_regressors
+        ).reshape(series_len, -1)
+        x = x * z
+
+    noise = np.random.default_rng(seed).normal(0, noise_scale, series_len)
+    # make observed response
+    y = np.matmul(x, coefs) + noise
+    return x, y, coefs
+
+
 def make_ts_multiplicative(series_len=200, seasonality=-1, coefs=None, regressor_relevance=0.0,
                            regressor_log_loc=0.0, regressor_log_scale=0.2,
                            regressor_log_cov=None,
                            noise_to_signal_ratio=1.0, regression_sparsity=0.5,
                            obs_val_base=1000, regresspr_val_base=1000,
                            trend_type='rw', rw_loc=0.001, rw_scale=0.1, seas_scale=.05,
                            response_col='y', seed=0):
-    """
+    """ [Deprecated] Module to generate multiplicative time-series with trend, seasonality and regression components
```

Comment:
we should update all instances of this function being used in tutorial notebooks

Reply:
They are not being used right in the current tutorials.  Once the other decomposed modules updated, I may bring back a demo using those new modules @steveyang90